File: 008-N-Base Notation Converter/NBaseNotationConverter.py
def DectoNBase(n, num):
    digits = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    if num == 0:
        return "0"
    result = ""
    while num:
        result = digits[num % n] + result
        num //= n
    return result

# Test Cases
print("Base 12, Number 718 =>", DectoNBase(12, 718))   # Output: 4BA
print("Base 21, Number 5678 =>", DectoNBase(21, 5678)) # Output: CI8

File: 008-N-Base Notation Converter/NBaseNotationConverter.cpp
#include<iostream>
#include<string>
#include<algorithm>

using namespace std;
string DectoNBase(int n, int num){
    string symbols = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    string result = "";

    if(n < 2 || n > 36){
        return "Error: Base must be between 2 and 36.";
    }

    if(num == 0){
        return "0";
    }

    while(num > 0){
        int remainder = num % n;
        result += symbols[remainder];
        num /= n;
    }

    reverse(result.begin(), result.end());
    return result;
}
int main(){
    int n, num;
    cout << "Enter the base (2-36): ";
    cin >> n;
    cout << "Enter the number: ";
    cin >> num;

    string result = DectoNBase(n, num);
    cout << "The number in base " << n << " is: " << result << endl;

    return 0;
}